src/providers/cryptoauthlib/hash.rs

```diff
@@ -33,4 +34,40 @@ impl Provider {
             }
         }
     }
+
+    pub(super) fn psa_hash_compare_internal(
+        &self,
+        op: psa_hash_compare::Operation,
+    ) -> Result<psa_hash_compare::Result> {
+        let op_hash = op.hash.to_vec();
+        let alg_len = op.alg.hash_length();
+        // calculate input hash
+        let op_compute = psa_hash_compute::Operation {
+            alg: op.alg,
+            input: op.input,
+        };
+        match self.psa_hash_compute_internal(op_compute) {
+            Ok(psa_hash_compute::Result { hash }) => {
+                // compare hashes length
+                if op_hash.len() != alg_len || hash.as_slice().len() != alg_len {
+                    let error = ResponseStatus::PsaErrorInvalidArgument;
+                    format_error!("Hash length comparison failed: ", error);
+                    return Err(error);
+                }
+                // compare hashes
+                if op_hash != hash.as_slice() {
+                    let error = ResponseStatus::PsaErrorInvalidSignature;
+                    format_error!("Hash comparison failed: ", error);
+                    return Err(error);
+                }
```

Comment:
Implementing the comparison in software is probably fine but I am wondering if the callers won't expect it to be done in the ATEC chip? Is it possible?
For example I am not sure if we care about the fact that this comparison won't be done in constant time, and if that might leak data to the original data being hashed. I am probably over-anxious

Reply:
Yes, this is true we violate the promise of having everything done in hardware. We are not aware of a possibility to offload this comparison to ATECC. So the conclusion is that most of hard job is done by hardware, while the "easy" tasks are left for software.
Please also note - whatever a message length is, the hash is always 32 bytes long, so the comparison is in an almost constant time, but still depending on host processor load, unfortunately.
I am not sure if I understand correctly the comment, but if ATECC could do the comparison, the message and the input hash would be transferred to it anyway. The only new information leaked with this implementation, and only locally (between the i2c ends i.e. CPU and ATECC) is the output hash calculated for the message. I hope this is acceptable.

Reply:
Thanks for the response. That is fine on my side!
I believe that for use-cases that care about efficiency, it's probably that the hash operations will be done in software anyway (as in, on the client side) with heavily optimised library. It is however useful to have those operations defined here, for convenience.